### PR TITLE
training loop refactor - move val loop 

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

We currently have the validation loop owned by the fit loop, i.e., 

`trainer.fit_loop.val_loop: EvaluationLoop`

It was introduced there in #8065.
However, the validation loop is never used in the fit loop. It is called only in the epoch loop, i.e.
`trainer.fit_loop.epoch_loop` calls the validation loop at the right interval. This PR moves the val loop 
from 

`trainer.fit_loop.val_loop` to 
`trainer.fit_loop.epoch_loop.val_loop`

This does not require a changelog entry, since we are developing these loops on master only at the moment.

Discussed with @tchaton and @carmocca so far.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/loops/epoch/__init__.py [hunk 1]
```diff
@@ -14,4 +14,4 @@
 
 from pytorch_lightning.loops.epoch.evaluation_epoch_loop import EvaluationEpochLoop  # noqa: F401
 from pytorch_lightning.loops.epoch.prediction_epoch_loop import PredictionEpochLoop  # noqa: F401
-from pytorch_lightning.loops.epoch.training_epoch_loop import TrainingEpochLoop  # noqa: F401
+# from pytorch_lightning.loops.epoch.training_epoch_loop import TrainingEpochLoop  # noqa: F401
```
Comment (awaelchli) [inline pytorch_lightning/loops/epoch/__init__.py:17]: @carmocca I'm trying to find an alternative here to resolve a cyclic import. 
seems pretty tricky to me

Comment (carmocca) [reply pytorch_lightning/loops/epoch/__init__.py]: Pushed 2 commits

1. grouping the imports to the largest level
2. Avoid the circular import by using `import * as *`

Tell me what you think of that

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/loops/epoch/__init__.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/loops/epoch/__init__.py]: I thought grouping to the largest level is one of the causes of the circular import

ok thanks

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
